### PR TITLE
Migrate store methods to use request.Context instead of context.Context

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Similar to https://github.com/mattermost/mattermost/pull/24514, this PR migrate more store methods to use `request.Context` instead of `context.Context`. This pushes `request.Context` further as an ubiquitously available context.

I'm fully aware that the code is not consistently using either `request.Context` or `request.CTX`. I've submitted a proposal in https://community.mattermost.com/core/pl/uxzkd5sj3j8n9cmifai415ijya that addresses this issue.

#### Ticket Link
NONE

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/store/sqlstore/context.go [hunk 1]
```diff
@@ -35,8 +35,8 @@ func RequestContextWithMaster(c request.CTX) request.CTX {
 	return c
 }
 
-// hasMaster is a helper function to check whether master DB should be selected or not.
-func hasMaster(ctx context.Context) bool {
+// HasMaster is a helper function to check whether master DB should be selected or not.
+func HasMaster(ctx context.Context) bool {
```
Comment (hanzei) [inline server/channels/store/sqlstore/context.go:39]: Needed for test

**Review by streamer45 — APPROVED** [review]
Nice! :+1:

File: server/channels/app/platform/license.go [hunk 2]
```diff
@@ -49,6 +49,8 @@ func (ps *PlatformService) License() *model.License {
 }
 
 func (ps *PlatformService) LoadLicense() {
+	c := request.EmptyContext(ps.logger)
```
Comment (fmartingr) [inline server/channels/app/platform/license.go:52]: I'm assuming this is safe to use around outside tests?

**Review by fmartingr — APPROVED** [review]
Awesome work refactoring this! 👏 👏

File: server/channels/app/platform/license.go (hunk 2 above)
Comment (hanzei) [reply server/channels/app/platform/license.go:52]: Yes, it is. There are plenty of places where it's used.

**Pull request merged**